### PR TITLE
DBZ-6455: Document Optimal MongoDB Oplog Config for Resiliency

### PR DESCRIPTION
**Pull request by ironakj**
Addresses/Closes DBZ-6455

### CODE REVIEW

File: documentation/modules/ROOT/pages/connectors/mongodb.adoc
Comment (ironakj) [inline documentation/modules/ROOT/pages/connectors/mongodb.adoc:1035]: @Naros @jpechane Can you help me with this? Should I just write "more than 3 GB would be optimum"

Comment (ironakj) [reply documentation/modules/ROOT/pages/connectors/mongodb.adoc:1035]: Also, I'll refine the language. This is just a draft for now.

Comment (jpechane) [reply documentation/modules/ROOT/pages/connectors/mongodb.adoc:1035]: @ironakj I think the text itself is fine. For the last sentence I 'd recommend something like - in suh situation if your average reaction time to Debezium failure is 2 hours then the oplog size should be at least 6 GB.

@roldanbob Could you please take a look at it when undrafted?

**Review by jpechane — APPROVED** [review]

**Comment by jpechane** [comment]
@roldanbob Could you please take a look at it? Thanks

**Review by roldanbob — CHANGES_REQUESTED** [review]
@ironakj I've made a few suggestions. If I've misunderstood anything or if you have questions about my proposed changes, give me a shout. Thanks!

File: documentation/modules/ROOT/pages/connectors/mongodb.adoc
Comment (ironakj) [reply documentation/modules/ROOT/pages/connectors/mongodb.adoc:1030]: Understood. In that case, adding it will create more confusion. 

I think here we can instead extend the previous statement to _"the connector *may* fails to start depending on your `snapshot.mode` config, and you must create a new connector."_

File: documentation/modules/ROOT/pages/connectors/mongodb.adoc
Comment (ironakj) [reply documentation/modules/ROOT/pages/connectors/mongodb.adoc:1034]: I think not everyone will be able to comprehend "time to react"  - a naive person might be just looking for how to recover from a failure and how to avoid one. Can you suggest how we can simplify it?

File: documentation/modules/ROOT/pages/connectors/mongodb.adoc
Comment (ironakj) [reply documentation/modules/ROOT/pages/connectors/mongodb.adoc:1036]: I think Replication Oplog Window is an important metric to look for in case of failure. I'll check how we can improve the above to add that - probably we should add that in the beginning itself. Any specific reason for removing it other than simplifying the text?

File: documentation/modules/ROOT/pages/connectors/mongodb.adoc
Comment (ironakj) [reply documentation/modules/ROOT/pages/connectors/mongodb.adoc:1032]: Will commit this.

File: documentation/modules/ROOT/pages/connectors/mongodb.adoc
Comment (roldanbob) [reply documentation/modules/ROOT/pages/connectors/mongodb.adoc:1036]: Yes, my intention was to simplify the text. The original sentence was a bit to dense, and its meaning was not entirely clear. The original link text refers generally to _Monitoring Oplog_ , but the link target is actually _Oplog GB/Hour_, so it seemed like the important takeaway for readers was that they should monitor replication metrics. I added a link in my [previous edit](https://github.com/debezium/debezium/pull/4598#discussion_r1235492479) to the Oplog GB/Hour metric, so readers who follow that link would land in the same _MongoDB Replication Metrics_ section that discusses the Replication Oplog Window.

I've added a couple of sentences to my original edit to suggest the specific metrics to track.

File: documentation/modules/ROOT/pages/connectors/mongodb.adoc
Comment (roldanbob) [reply documentation/modules/ROOT/pages/connectors/mongodb.adoc:1034]: Edited.

File: documentation/modules/ROOT/pages/connectors/mongodb.adoc
Comment (roldanbob) [reply documentation/modules/ROOT/pages/connectors/mongodb.adoc:1030]: Edited.

**Comment by ironakj** [comment]
Based on the review I've done the changes, @roldanbob  Could you review it again?  Also, let's get Jiri's opinion about the "snapshot-mode" once you're done with the review (as I am not completely sure how it works)

One more thing, will it help if add the exact error usually faced when this situation, as most people will come and look for the actual error they face? 

Should we extend "the connector might fail to start" to "the connector might fail to start, stating Resume point not available" 

or

Should we create a new section at the bottom (within common issues) we can describe the error `Resume Point No Longer Available` and suggest checking "Optimum Oplog Config"

**Comment by jpechane** [comment]
@jcechace Could you please provide an input if needed so the PR can be finished? Thanks

**Comment by roldanbob** [comment]
> ... will it help to add the exact error usually faced in this situation, as most people will come and look for the actual error they face?
> 
> Should we extend "the connector might fail to start" to "the connector might fail to start, stating Resume point not available" ...  or ... create a new section at the bottom (within common issues) [to] describe the error `Resume Point No Longer Available` and suggest checking "Optimum Oplog Config"

Both? I think both is the correct answer here. While we typically want to avoid redundancy, in this case, I think it makes sense, because, as you suggest, customers who encounter an error might look first to the "issues" section, but we should also provide the information up front so that users might avoid the error, or at least be cognizant of the scenario that could lead to it.

**Comment by ironakj** [comment]
@roldanbob Understood, will make the changes accordingly and request re-review

**Comment by ironakj** [comment]
@roldanbob Changes done, kindly re-review.

**Review by jcechace — APPROVED** [review]
LGTM

One note -- the usually encountered error message when resume token is no longer valid is just that "the usual". There are few variants of that message one is by far the most common one. I'm in favour of keeping this as is in order not to pollute the docs with every possible error message MongoDB is capable producing.

**Comment by jcechace** [comment]
@roldanbob  once this gets your final approval I will merge.

**Comment by ironakj** [comment]
> @roldanbob once this gets your final approval I will merge.

Understood, the idea is to give a gist of what to expect when a connector is paused for a while and how Oplog retention is critical to avoid data loss and config that helps achieve the same. As maintenance downtime can be imminent, and due to different data loads there isn't one fix that can work everywhere, thus it's better for the user to understand this event before deploying.

File: documentation/modules/ROOT/pages/connectors/mongodb.adoc
Comment (roldanbob) [inline documentation/modules/ROOT/pages/connectors/mongodb.adoc:1991]:
```suggestion
The preceding exception indicates that the entry that corresponds to the connector's resume token is no longer present in the oplog. 
Because the oplog no longer contains the last offset that the connector processed, the connector cannot resume streaming.
```

File: documentation/modules/ROOT/pages/connectors/mongodb.adoc
Comment (roldanbob) [inline documentation/modules/ROOT/pages/connectors/mongodb.adoc:1993]:
```suggestion
You can use either of the following options to recover from the failure:

* Delete the failed connector, and create a new connector with the same configuration but with a different connector name.

* Pause the connector and then remove offsets, or change the offset topic.
```

File: documentation/modules/ROOT/pages/connectors/mongodb.adoc
Comment (roldanbob) [inline documentation/modules/ROOT/pages/connectors/mongodb.adoc:1995]:
```suggestion
To help prevent failures related to missing resume tokens, xref:mongodb-optimal-oplog-config[optimize configuration of the oplog].
```

**Review by roldanbob — CHANGES_REQUESTED** [review]
@ironakj  I've suggested a few changes. As a general note, be consistent in using asciidoc format for links. You've used a mix of markdown and asciidoc formatting.
Thanks!

**Comment by ironakj** [comment]
> @ironakj I've suggested a few changes. As a general note, be consistent in using asciidoc format for links. You've used a mix of markdown and asciidoc formatting. Thanks!

Understood. Applied the suggested changes.

**Review by roldanbob — APPROVED** [review]
Not sure if this is waiting on me, but based on incorporation of the feedback from my last review, I don't have any further changes to suggest. LGTM.

**Review by jcechace — APPROVED** [review]

**Pull request merged**
